management/src/clusterm/manager/client.go

```diff
@@ -93,55 +99,67 @@ func (c *Client) doGet(rsrc string) ([]byte, error) {
 }
 
 // PostNodeCommission posts the request to commission a node
-func (c *Client) PostNodeCommission(nodeName, extraVars string) error {
-	return c.doPost(fmt.Sprintf("%s/%s", PostNodeCommissionPrefix, nodeName), extraVars, nil)
+func (c *Client) PostNodeCommission(nodeName string, flags ActionFlags) error {
+	if !IsValidHostGroup(flags.HostGroup) {
+		return errored.Errorf("host-group is a mandatory parameter and is not specified %s", flags.HostGroup)
+	}
+
+	req := &APIRequest{
+		HostGroup: flags.HostGroup,
+	}
+	return c.doPost(fmt.Sprintf("%s/%s", PostNodeCommissionPrefix, nodeName), flags.ExtraVars, req)
 }
 
 // PostNodesCommission posts the request to commission a set of nodes
-func (c *Client) PostNodesCommission(nodeNames []string, extraVars string) error {
+func (c *Client) PostNodesCommission(nodeNames []string, flags ActionFlags) error {
+	if !IsValidHostGroup(flags.HostGroup) {
+		return errored.Errorf("host-group is a mandatory parameter and is not specified %s", flags.HostGroup)
+	}
+
 	req := &APIRequest{
-		Nodes: nodeNames,
+		Nodes:     nodeNames,
+		HostGroup: flags.HostGroup,
 	}
-	return c.doPost(PostNodesCommission, extraVars, req)
+	return c.doPost(PostNodesCommission, flags.ExtraVars, req)
 }
 
 // PostNodeDecommission posts the request to decommission a node
-func (c *Client) PostNodeDecommission(nodeName, extraVars string) error {
-	return c.doPost(fmt.Sprintf("%s/%s", PostNodeDecommissionPrefix, nodeName), extraVars, nil)
+func (c *Client) PostNodeDecommission(nodeName string, flags ActionFlags) error {
```

Comment:
This is similar to comment above regarding Commission method. Decommission method shall continue to take only node name and extra-vars as argument from the caller. This keep the public interface contract clear for the caller. Same shall hold for respective Maintenance, Globals and multi-host methods below.